numpy/lib/function_base.py

```diff
@@ -2119,8 +2119,8 @@ def _create_arrays(broadcast_shape, dim_sizes, list_of_core_dims, dtypes,
 @set_module('numpy')
 class vectorize:
     """
-    vectorize(pyfunc, otypes=None, doc=None, excluded=None, cache=False,
-              signature=None)
+    vectorize(pyfunc=np._NoValue, otypes=None, doc=None, excluded=None,
```

Comment:
Maybe time to replace this with something like "Create a vectorized version of a function, able to process arrays" -- after all, the signature is always shown by python help and the docs.

Reply:
Agreed. Strictly speaking, I don't even think np.vectorize counts as a generalized function class.

Reply:
> Is there a reason not to use `None`?

Using would make ```@vectorize(None)``` a legal call. It's not the worst thing, but it's strange and I'd prefer avoiding it. There was a bit of controversy over this in the old PR, btw.

Reply:
Looking at this again, I think `None` is actually fine here...

Reply:
>Looking at this again, I think None is actually fine here...

I just don't want to risk some edge case where somebody accidentally passes None.

Reply:
Eric preferred to not use None in the other PR.  Personally, I still prefer `None` slightly, `NoValue` seems a bit too smart for no reason I can think of where users might pass `None` accidentally.